indico/util/date_time.py

```diff
@@ -229,6 +229,66 @@ def format_human_date(dt, format='medium', locale=None):
         return format_date(dt, format, locale=locale)
 
 
+def _format_pretty_datetime(dt, locale, tzinfo, formats):
+    locale = get_current_locale() if not locale else parse_locale(locale)
+
+    if tzinfo:
+        if dt.tzinfo:
+            dt = dt.astimezone(tzinfo)
+        else:
+            dt = tzinfo.localize(dt).astimezone(tzinfo)
+
+    today = (now_utc(False).astimezone(tzinfo) if tzinfo else now_utc(False)).replace(hour=0, minute=0)
+    diff = (dt - today).total_seconds() / 86400.0
+    mapping = [(-6, 'other'), (-1, 'last_week'), (0, 'last_day'),
+               (1, 'same_day'), (2, 'next_day'), (7, 'next_week'),
+               (None, 'other')]
+
+    fmt = next(formats[key] for delta, key in mapping if delta is None or diff < delta)
+    fmt = fmt.format(date_fmt=locale.date_formats['short'], time_fmt=locale.time_formats['short'])
+    return _format_datetime(dt, fmt, tzinfo, locale)
+
+
+def format_pretty_date(dt, locale=None, tzinfo=None):
+    """Format a date in a pretty way using relative units if possible.
+
+    :param dt: a date or datetime object. if a date is provided, its
+               time is assumed to be midnight
+    :param locale: the locale to use for formatting
+    :param tzinfo: the timezone to use
+    """
+    if not isinstance(dt, datetime):
+        dt = datetime.combine(dt, dt_time())
+    return _format_pretty_datetime(dt, locale, tzinfo, {
```

Comment:
None I know about - everyone has their own preferences on how to format relative dates.